pkg/vfs/vfs.go

```diff
@@ -94,6 +94,9 @@ type Indexer interface {
 
 	// CreateFileDoc creates and add in the index a new file document.
 	CreateFileDoc(doc *FileDoc) error
+	// CreateNamedFileDoc creates and add in the index a new file document with
+	// its id already set.
+	CreateNamedFileDoc(doc *FileDoc) error
```

Comment:
Rather than adding a new method in this interface (which is already pretty big...), why not handling the case of a document with a non-empty `ID()` in the `CreateFileDoc()` method directly ?